bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/scalarweb/models/ScalarWebService.java

```diff
@@ -0,0 +1,402 @@
+                if (mthdTransport == null) {
+                    logger.debug("No transport for {} with protocols: {}", request, protocols);
+                    return new ScalarWebResult(HttpStatus.INTERNAL_SERVER_ERROR_500,
+                            "No transport for " + request + " with protocols: " + protocols);
+                } else {
+                    logger.debug("Execution of {} is using a different protocol {} than the service {}", request,
+                            mthdTransport.getProtocolType(), transport.getProtocolType());
+                    return mthdTransport.execute(request, options);
+                }
+            }
+        }
+    }
+
+    /**
+     * Returns the label for a given service name
+     * 
+     * @param serviceName a non-null, non-empty service name
+     * @return the label for the service
+     */
+    private static String labelFor(final String serviceName) {
+        Validate.notEmpty(serviceName, "serviceName cannot be empty");
+        switch (serviceName) {
+            case ACCESSCONTROL:
+                return "Access Control";
+            case APPCONTROL:
+                return "Application Control";
+            case AUDIO:
+                return "Audio";
+            case AVCONTENT:
+                return "A/V Content";
+            case BROWSER:
+                return "Browser";
+            case CEC:
+                return "CEC";
+            case CONTENTSHARE:
+                return "Content Share";
+            case ENCRYPTION:
+                return "Encryption";
+            case GUIDE:
+                return "Guide";
+            case ILLUMINATION:
+                return "Illumination";
+            case SYSTEM:
+                return "System";
+            case VIDEOSCREEN:
+                return "Video Screen";
+            default:
+                return serviceName;
+        }
+    }
+
+    /**
+     * Returns a map of service names to their label
+     * 
+     * @return a non-null, non-empty map of service names to service labels
+     */
+    public static Map<String, String> getServiceLabels() {
+        return Collections.unmodifiableMap(new HashMap<String, String>() {
+            private static final long serialVersionUID = 5934100497468165317L;
+            {
+                put(ACCESSCONTROL, labelFor(ACCESSCONTROL));
+                put(APPCONTROL, labelFor(APPCONTROL));
+                put(AUDIO, labelFor(AUDIO));
+                put(AVCONTENT, labelFor(AVCONTENT));
+                put(BROWSER, labelFor(BROWSER));
+                put(CEC, labelFor(CEC));
+                put(CONTENTSHARE, labelFor(CONTENTSHARE));
+                put(ENCRYPTION, labelFor(ENCRYPTION));
+                put(GUIDE, labelFor(GUIDE));
+                put(ILLUMINATION, labelFor(ILLUMINATION));
+                put(SYSTEM, labelFor(SYSTEM));
+                put(VIDEOSCREEN, labelFor(VIDEOSCREEN));
+            }
+        });
```

Comment:
Make this a static field

Reply:
Can't for the life of me remember why I did it this way (beyond remembering that I did for some reason).  Making it static however